File: my-folder/0125-valid-palindrome/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();

    return check(0, s.length - 1, s);
};

function check(i, j, s) {
    if (s.length <= 1 || i >= s.length / 2) {
        return true;
    }

    return (s[i] === s[j]) && check(i + 1, j - 1, s);
}

File: my-folder/0125-valid-palindrome/solution.rs
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let s = s.to_ascii_lowercase()
                 .chars()
                 .filter(|&c| c.is_ascii_alphanumeric())
                 .collect::<Vec<char>>();

        check(0, s.len() - 1, s)
    }


}
    fn check(i: usize, j: usize, s: Vec<char>) -> bool {
        if s.len() <= 1 || i >= s.len() / 2 {
            return true;
        }

        (s[i] == s[j]) && check(i + 1, j - 1, s)
    }
